frontend/system/src/Entities/ListView.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useEffect, useState } from 'react';
+import { useTranslation } from '@common/intl';
+import {
+  AppBarContentPortal,
+  DataTable,
+  NothingHere,
+  TableProvider,
+  createTableStore,
+  useColumns,
+  SearchToolbar,
+  ToggleButton,
+} from '@common/ui';
+import { useBreadcrumbs, useQueryParamsState } from '@common/hooks';
+import { useEntities } from '../api';
+import { ToggleButtonGroup } from '@mui/material';
+
+export const ListView = () => {
+  const t = useTranslation('system');
+  const { setSuffix } = useBreadcrumbs();
+
+  useEffect(() => {
+    setSuffix('Browse');
+  }, []);
+
+  const { filter, queryParams, updatePaginationQuery, updateSortQuery } =
+    useQueryParamsState({
+      initialSort: { key: 'description', dir: 'asc' },
+    });
+
+  const { sortBy, page, offset, first } = queryParams;
+
+  const columns = useColumns(
+    [
+      { key: 'code', label: 'label.code', width: 200, sortable: false },
+      { key: 'description', label: 'label.description', width: 1000 },
+      { key: 'type', label: 'label.type', sortable: false, width: 200 },
+    ],
+    { sortBy: sortBy, onChangeSortBy: updateSortQuery },
+    [sortBy, updateSortQuery]
+  );
+
+  const [categories, setCategories] = useState<string[]>(['drug']);
+
+  const searchFilter = filter.filterBy?.['search'];
+  const search = typeof searchFilter === 'string' ? searchFilter : '';
+
+  const { data, isError, isLoading } = useEntities({
+    filter: {
+      categories,
+      description: search,
+      orderBy: {
+        field: sortBy.key,
+        descending: sortBy.isDesc,
+      },
+    },
+    first,
+    offset,
+  });
+
+  const updateCategories = (category: string) => {
+    if (categories.includes(category)) {
+      setCategories(categories.filter(c => c !== category));
+    } else {
+      setCategories([...categories, category]);
+    }
+  };
+
+  const entities = data?.data ?? [];
+
+  const pagination = {
+    page,
+    offset,
+    first,
+    total: data?.totalLength,
+  };
+
+  return (
+    <>
+      <TableProvider createStore={createTableStore}>
+        <AppBarContentPortal
+          sx={{
+            paddingBottom: '16px',
+            flex: 1,
+            display: 'flex',
+            justifyContent: 'space-between',
+          }}
+        >
+          <SearchToolbar filter={filter} />
```

Comment:
This search control is still experiencing that weird behaviour where we click the `x` button and the box just get smaller. I thought it was fixed in notify but maybe not? Can you check?

Reply:
Thanks for spotting this! Turns out it was... half fixed in Notify, but there was some kind of race condition between the onClick firing as expected, or the focus shifting away from input before it could get there. I've [updated it to use `onMouseDown`](https://github.com/msupply-foundation/unified-codes/pull/434/commits/f0ef8b2782063bb30a1cc41979117c85c54199db) instead, which appears to be working reliably! Will make a PR for Notify (and HSH if the same thing is there?) now!